node/primitives/src/lib.rs

```diff
@@ -255,6 +266,9 @@ pub enum InvalidCandidate {
 	CodeHashMismatch,
 	/// Validation has generated different candidate commitments.
 	CommitmentsHashMismatch,
+	/// The worker has died during validation of a candidate. See
+	/// [`InvalidCandidate::AmbiguousWorkerDeath`].
+	AmbiguousWorkerDeath,
 }
```

Comment:
Hmm, interesting. Before I jump back into this though, do you think this PR is worth pursuing further? Are the metrics buckets useful to us?

[TBH I thought this would be an easy change, and wanted to do minimal refactoring just to get more metrics buckets (I guess I got a bit carried away with `PrepareError`). My point being that I don't really want to do more reworking of the errors. 😛]

Reply:
I think these are too many metrics. What should an operator do with all these metrics? They want to know if the node is running correctly and if not, they should be notified. So, `Valid`, `Invalid`, `Invalid_Your_Validator_Has_Issues` should be enough IMO, while the latest one would be the thing that operators use to add some kind of alert. When this metric starts rising quickly, something is probably wrong with their node. However, for a detailed analysis logs will be required.

Reply:
I agree with Basti

Reply:
Fair enough! I can close this PR, should I close the related issue as well?

Reply:
+1 metrics should provide a birds eye view for edge cases or things not working correctly in general. Sometimes having more details (more metrics) makes sense but I've only seen that useful for performance tracking.